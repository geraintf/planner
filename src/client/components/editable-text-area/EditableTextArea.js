import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './EditableTextArea.less';

export default class EditableTextArea extends Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    onKeyPress: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);

    this.textArea = React.createRef();

  }

  componentDidMount() {
    this.textArea.current.focus();
  }

  render() {
    const {
      text,
      onBlur,
      onChange,
      onKeyPress
    } = this.props;

    return (
      <div className="editable-text-area">
        <textarea
          ref={this.textArea}
          value={text}
          onBlur={onBlur}
          onChange={onChange}
          onKeyPress={onKeyPress}
        />
      </div>
    );
  }
}
