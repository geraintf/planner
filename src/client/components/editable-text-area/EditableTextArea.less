.editable-text-area {
  textarea {
    height: 55px;
    resize: none;
    width: 100%;
  }
}