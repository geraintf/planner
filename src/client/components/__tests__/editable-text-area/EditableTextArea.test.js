import React from 'react';
import { mount } from 'enzyme';

import EditableTextArea from '../../editable-text-area/EditableTextArea';

describe('EditableTextArea', () => {
  let output;
  let props;

  beforeEach(() => {
    props = {
      text: "test text",
      onBlur: () => {},
      onChange: () => {},
      onKeyPress: () => {}
    };
    output = mount(<EditableTextArea {...props} />);
  });

  it('should render correctly', () => {
    expect(output).toMatchSnapshot();
  });

  it('should have focus on mounting', () => {
    const textArea = output.find('textarea');
    const focusedElement = document.activeElement;

    expect(textArea.matchesElement(focusedElement)).toEqual(true);
  });
});
