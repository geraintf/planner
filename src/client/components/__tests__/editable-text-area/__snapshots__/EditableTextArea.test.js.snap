// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTextArea should render correctly 1`] = `
<EditableTextArea
  onBlur={[Function]}
  onChange={[Function]}
  onKeyPress={[Function]}
  text="test text"
>
  <div
    className="editable-text-area"
  >
    <textarea
      onBlur={[Function]}
      onChange={[Function]}
      onKeyPress={[Function]}
      value="test text"
    />
  </div>
</EditableTextArea>
`;
